perma_web/perma/tasks.py

```diff
@@ -2503,3 +2512,365 @@ def confirm_file_uploaded_to_internet_archive(file_id, attempts=0):
+
+    This may be too blunt an instrument; we may need to introduce a delay in the confirmation
+    task itself, sleeping between each retry, but we want to try this first: if we can, we want
+    to avoid having sleeping-but-active celery tasks.
+    """
+    file_ids = InternetArchiveFile.objects.filter(
+                status='deletion_attempted'
+            ).exclude(
+                item_id__in=[
+                    'daily_perma_cc_2022-07-25',
+                    'daily_perma_cc_2022-07-21',
+                    'daily_perma_cc_2022-07-20',
+                    'daily_perma_cc_2022-07-19'
+                ]
+            ).values_list(
+                'id', flat=True
+            )[:limit]
+
+    queued = 0
+    for file_id in file_ids.iterator():
+        confirm_file_deleted_from_daily_item.delay(file_id)
+        queued = queued + 1
+    logger.info(f"Queued the file deleted confirmation task for {queued} InternetArchiveFiles.")
+
+
+
+@shared_task
+def queue_internet_archive_uploads_for_date(date_string, limit=None):
+    """
+    Queue upload tasks for all currently-eligible Links created on a given day,
+    if we have not yet attempted to upload them to a "daily" Item.
+    """
+
+    # Get all Links we think should have been uploaded to IA,
+    # and then filter out the ones that have already been uploaded
+    # to a "daily" item.
+    #
+    # Do so with our own SQL because the SQL generated from the
+    # intuitive ORM query proved to be impossibly slow.
+    # links = Link.objects.visible_to_ia().filter(
+    #     creation_timestamp__date=InternetArchiveItem.date(date_string)
+    # ).exclude(
+    #     internet_archive_items__span__isempty=False
+    # )
+    sql = '''
+        WITH links_with_daily_items AS (
+            SELECT
+              perma_link.guid
+            FROM
+              perma_link
+              INNER JOIN perma_internetarchivefile ON
+                perma_link.guid = perma_internetarchivefile.link_id
+              INNER JOIN perma_internetarchiveitem ON
+                perma_internetarchivefile.item_id = perma_internetarchiveitem.identifier
+            WHERE
+              isempty(
+                perma_internetarchiveitem.span
+              ) = False
+        )
+
+        SELECT
+          perma_link.guid
+        FROM
+          perma_link
+          LEFT JOIN links_with_daily_items ON
+               perma_link.guid = links_with_daily_items.guid
+        WHERE
+          links_with_daily_items.guid IS NULL
+          AND (
+            (perma_link.creation_timestamp AT TIME ZONE 'UTC')::DATE = %s
+          )
+          AND (
+            perma_link.user_deleted = False AND perma_link.is_private = False AND perma_link.is_unlisted = False AND perma_link.cached_can_play_back = True
+          )
+    '''
+
+    # Get all links we think should have been uploaded to IA that are not yet associated with a daily InternetArchiveItem.
+    if limit:
+        sql += "LIMIT %s"
```

Comment:
```suggestion
        sql += " LIMIT %s"
```

Minor but add whitespace to guard against appending this token directly adjacent to a previous one leading to invalid SQL.